### PR TITLE
CICD Pipeline

### PR DESCRIPTION
**Pull request by pdittaro**
This work captures the initial end-to-end CD pipeline into the development environment.

On push to main branch:
- Build API, Frontend, and Migrate Images and push them to the internal Github Container Registry
- Import the Image Stream in OpenShift to reference the latest version of these images in preparation for,
- Tagging the images in the registry with the specific 'Github Action Run Number'
- Then, create a workflow dispatch in the bcgov-c gitops repository which includes the image names to be updated
- A seperate github action updates the appropriate namespace patches to reference these new images
- Once this has been commited (by the action), ArgoCD will pick up the changes and apply the latest images, including running a database migration job before the Sync action

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+name: build
+
+on: 
+  workflow_call:
```
Comment (dhaselhan) [inline .github/workflows/deploy.yml:4]: What does this do?

**Review by dhaselhan — APPROVED** [review]
LGTM 👍

**Review by mhuseinov — COMMENTED** [review]
Would be nice to get some readme file that could provide some overview to the deployment structure

**Review by mhuseinov — APPROVED** [review]

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (pdittaro) [reply .github/workflows/deploy.yml:4]: Only runs when something explictly calls it, see the cd.yaml file

Comment (pdittaro) [reply .github/workflows/deploy.yml:4]: If you mean the environment input, right now its specifying which dispatch event to trigger on the child repo

**Pull request merged**